src/bot/commands/lastfm/fm.ts

```diff
@@ -249,32 +289,100 @@ export class LastFm extends Command {
                             files: [await image.getBufferAsync(Jimp.MIME_PNG)]
                         });
                     });
+                }  else {
+                    input.channel.send('Please connect your lastfm account using `lastfm set <username>`');
                 }
                 break;
             case 'artistchart':
+                if (user && user != '') {
+                    let currentTime = (Math.floor(new Date().getTime()/1000.0));
+                    let from = currentTime-315569260;
+                    if (input.getArgument('user')) {
+                        switch(user) {
+                            case 'week':
+                            case 'weekly':
+                                from = currentTime-604800;
+                                break;
+                            case 'month':
+                            case 'monthly':
+                                from = currentTime-2629743;
+                                break;
+                            case 'year':
+                            case 'yearly':
+                                from = currentTime-31556926;
+                                break;
+                            case 'all':
+                            case 'alltime':
+                                from = currentTime-315569260;
+                                break;
+                            default:
+                                from = currentTime-315569260;
+                        }
+                    }
+
+                    let requestURL = request((lastfmURL + 'user.getWeeklyArtistChart' + '&user=' + db.lastfmId + '&from=' + from.toString() + '&to=' + currentTime.toString() + '&api_key= ' + key + '&format=json'), async (error: any, response: Response, body: any) => {
+                        if (error) {
+                            input.channel.send(`${Emoji.ERROR}  Connection error! Unable to retrieve lastfm data.`);
+                            return;
+                        }
+
+                        let parsed = JSON.parse(body);
+
+                        if (parsed.weeklyartistchart == undefined) {
+                            input.channel.send(`${Emoji.ERROR}  Connection error! Unable to retrieve lastfm data.`);
+                            return;
+                        }
+
+                        let image : Jimp = await Jimp.read(pub('images/blankchart.png')) as Jimp;
+
+                        let count = 0;
+                        for (let i = 0; i < 3; i++) {
+                            for (let j = 0; j < 3; j++) {
```

Comment:
I understand that i and j are commonly used in for loops, but personally I would use more meaningful variable names. Reading this I don't entirely understand what i and j are supposed to represent.

Reply:
`x` and `y` might be good alternatives for a grid like this.